nexus/src/external_api/shared.rs

```diff
@@ -40,12 +40,56 @@ pub const MAX_ROLE_ASSIGNMENTS_PER_RESOURCE: usize = 64;
 /// access to this resource.
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize, JsonSchema)]
 #[schemars(rename = "{AllowedRoles}Policy")]
-pub struct Policy<AllowedRoles: serde::de::DeserializeOwned> {
+pub struct Policy<AllowedRoles: serde::de::DeserializeOwned + std::clone::Clone>
```

Comment:
I wouldn't think you'd need `Clone` here provided it's on the impl block below.  If you can drop it, I think you can drop some of the other changes in this PR that added a `Clone` bound.